shell/src/main/java/alluxio/cli/TestRunner.java

```diff
@@ -122,47 +141,115 @@ private int runTests() throws Exception {
+                Constants.ANSI_RED, getWorkerAddressString(k), 4, Constants.ANSI_RESET);
+          }
+        });
+      }
     }
     return failed;
   }
 
-  /**
-   * Runs a single test given operation, read and write type.
-   *
-   * @param opType operation type
-   * @param readType read type
-   * @param writeType write type
-   * @return 0 on success, 1 on failure
-   */
-  private int runTest(OperationType opType, ReadType readType, WriteType writeType,
-      FileSystemContext fsContext) {
-    AlluxioURI filePath =
-        new AlluxioURI(String.format("%s/%s_%s_%s", mDirectory, opType, readType, writeType));
+    /**
+     * Runs a single test given operation, read and write type.
+     *
+     * @param opType operation type
+     * @param readType read type
+     * @param writeType write type
+     * @return 0 on success, 1 on failure
+     */
+  private int runTest(
+      OperationType opType, ReadType readType, WriteType writeType,
+      FileSystemContext fsContext, @Nullable WorkerNetAddress workerAddress) {
+    final AlluxioURI filePath;
+    if (workerAddress == null) {
+      filePath = new AlluxioURI(
+          String.format("%s/%s_%s_%s", mDirectory, opType, readType, writeType));
+    } else {
+      String workerAddressString = getWorkerAddressString(workerAddress);
+      filePath = new AlluxioURI(
+          String.format("%s/%s/%s_%s_%s", mDirectory, workerAddressString, opType, readType,
+              writeType));
+    }
 
     boolean result = true;
     switch (opType) {
       case BASIC:
         result = RunTestUtils.runExample(
-          new BasicOperations(filePath, readType, writeType, fsContext));
+          new BasicOperations(filePath, readType, writeType, fsContext, workerAddress));
         break;
       case BASIC_NON_BYTE_BUFFER:
         result = RunTestUtils.runExample(
-          new BasicNonByteBufferOperations(filePath, readType, writeType, true, 20, fsContext));
+          new BasicNonByteBufferOperations(
+              filePath, readType, writeType, true, 20, fsContext, workerAddress));
         break;
       default:
         System.out.println("Unrecognized operation type " + opType);
     }
     return result ? 0 : 1;
   }
+
+  /**
+   * Parses worker address param.
+   */
+  public static class WorkerAddressConverter implements IStringConverter<WorkerNetAddress> {
+    @Override
+    public WorkerNetAddress convert(String s) {
+      if (s.contains(":")) {
+        String[] components = s.split(":");
+        Preconditions.checkState(components.length == 2);
+        return WorkerNetAddress.newBuilder().setHost(components[0])
+            .setRpcPort(Integer.parseInt(components[1])).build();
+      } else {
+        return WorkerNetAddress.newBuilder().setHost(s).build();
+      }
+    }
+  }
+
+  private String getWorkerAddressString(WorkerNetAddress workerAddress) {
+    return workerAddress.getRpcPort() == 0 ? workerAddress.getHost() :
```

Comment:
i think HDFS doesn't allow `:` in the path. Let's replace the colon with something like underscore?

Reply:
good catch addressed